x-pack/apm-server/sampling/processor.go

```diff
@@ -397,6 +367,7 @@ func (p *Processor) Run() error {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-p.stopping:
+			<-time.After(shutdownGracePeriod)
```

Comment:
```suggestion
			select {
			case <-ctx.Done():
				return ctx.Err()
			case <-time.After(shutdownGracePeriod):
			}
```

This would allow the processor to exit faster than the grace period, if the final flush completes faster.